catalog/resource_sql_table.go

```diff
@@ -0,0 +1,383 @@
+	StorageLocation       string            `json:"storage_location,omitempty" tf:"suppress_diff"`
+	StorageCredentialName string            `json:"storage_credential_name,omitempty" tf:"force_new"`
+	ViewDefinition        string            `json:"view_definition,omitempty"`
+	Comment               string            `json:"comment,omitempty"`
+	Properties            map[string]string `json:"properties,omitempty" tf:"computed"`
+	ClusterID             string            `json:"cluster_id,omitempty" tf:"computed"`
+
+	exec common.CommandExecutor
+}
+
+type SqlTablesAPI struct {
+	client  *common.DatabricksClient
+	context context.Context
+}
+
+func NewSqlTablesAPI(ctx context.Context, m any) SqlTablesAPI {
+	return SqlTablesAPI{m.(*common.DatabricksClient), context.WithValue(ctx, common.Api, common.API_2_1)}
+}
+
+func (a SqlTablesAPI) getTable(name string) (ti SqlTableInfo, err error) {
+	err = a.client.Get(a.context, "/unity-catalog/tables/"+name, nil, &ti)
+	return
+}
+
+func (ti *SqlTableInfo) FullName() string {
+	return fmt.Sprintf("%s.%s.%s", ti.CatalogName, ti.SchemaName, ti.Name)
+}
+
+type SqlTables struct {
+	Tables []SqlTableInfo `json:"tables"`
+}
+
+// These properties are added automatically
+// If we do not customize the diff using these then terraform will constantly try to remove them
+// `properties` is essentially a "partially" computed field
+// This needs to be replaced with something a bit more robust in the future
+func sqlTableIsManagedProperty(key string) bool {
+	managedProps := map[string]bool{
+		"delta.lastCommitTimestamp":                                true,
+		"delta.lastUpdateVersion":                                  true,
+		"delta.minReaderVersion":                                   true,
+		"delta.minWriterVersion":                                   true,
+		"view.catalogAndNamespace.numParts":                        true,
+		"view.catalogAndNamespace.part.0":                          true,
+		"view.catalogAndNamespace.part.1":                          true,
+		"view.query.out.col.0":                                     true,
+		"view.query.out.numCols":                                   true,
+		"view.referredTempFunctionsNames":                          true,
+		"view.referredTempViewNames":                               true,
+		"view.sqlConfig.spark.sql.hive.convertCTAS":                true,
+		"view.sqlConfig.spark.sql.legacy.createHiveTableByDefault": true,
+		"view.sqlConfig.spark.sql.parquet.compression.codec":       true,
+		"view.sqlConfig.spark.sql.session.timeZone":                true,
+		"view.sqlConfig.spark.sql.sources.commitProtocolClass":     true,
+		"view.sqlConfig.spark.sql.sources.default":                 true,
+		"view.sqlConfig.spark.sql.streaming.stopTimeout":           true,
+	}
+	return managedProps[key]
+}
+
+func (ti *SqlTableInfo) initCluster(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) (err error) {
```

Comment:
can you add package-level mutex here? you'll have hundreds of tables created from the same state and they'll be all racing for creation/start of a cluster. it's not blocking this PR and could be done as a follow-up, though.

Reply:
This was copied directly from sql_permissions and there is already a mutex in the method which creates the cluster: 

```go
func (a ClustersAPI) GetOrCreateRunningCluster(name string, custom ...Cluster) (c ClusterInfo, err error) {
	getOrCreateClusterMutex.Lock()
	defer getOrCreateClusterMutex.Unlock()
```

I have locally tested creating a significant # of tables and the existing mutex works.

Reply:
Seems the new clusters API doesnt have an equivalent for  `GetOrCreateRunningCluster` method so when i reimplement this i will add the mutex